engines/dm/dungeonman.cpp

```diff
@@ -0,0 +1,1672 @@
+/* ScummVM - Graphic Adventure Engine
+*
+* ScummVM is the legal property of its developers, whose names
+* are too numerous to list here. Please refer to the COPYRIGHT
+* file distributed with this source distribution.
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License
+* as published by the Free Software Foundation; either version 2
+* of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*
+*/
+
+/*
+* Based on the Reverse Engineering work of Christophe Fontanel,
+* maintainer of the Dungeon Master Encyclopaedia (http://dmweb.free.fr/)
+*/
+
+#include "common/file.h"
+#include "common/memstream.h"
+
+#include "dungeonman.h"
+#include "timeline.h"
+#include "champion.h"
+#include "group.h"
+#include "movesens.h"
+#include "projexpl.h"
+
+namespace DM {
+
+void DungeonMan::mapCoordsAfterRelMovement(Direction dir, int16 stepsForward, int16 stepsRight, int16 &posX, int16 &posY) {
+	posX += _vm->_dirIntoStepCountEast[dir] * stepsForward;
+	posY += _vm->_dirIntoStepCountNorth[dir] * stepsForward;
+	turnDirRight(dir);
+	posX += _vm->_dirIntoStepCountEast[dir] * stepsRight;
+	posY += _vm->_dirIntoStepCountNorth[dir] * stepsRight;
+}
+
+void DungeonMan::setupConstants() {
+	ObjectInfo objectInfo[180] = { // @ G0237_as_Graphic559_ObjectInfo
+		/* { Type, ObjectAspectIndex, ActionSetIndex, AllowedSlots } */
+		ObjectInfo(30,  1,  0, 0x0500),   /* COMPASS        Pouch/Chest */
```

Comment:
Yes, but I don't manage to work properly once the DAT file is in place (see Access engine, I'm stuck because of it). So I considered creating a DAT file later, when more variants and games are supported